packages/sessions-sdk-react/src/components/token-list.tsx

```diff
@@ -150,6 +100,81 @@ export const TokenList = ({
   }
 };
 
+type TokenItemProps = {
+  token: Token;
+  onPressSend?: (token: Token) => void;
+  onPressToken?: (token: Token) => void;
+};
+
+const TokenItem = ({ token, onPressSend, onPressToken }: TokenItemProps) => {
+  const { mint, amountInWallet, decimals, image, name } = token;
+  const amountAsString = amountToString(amountInWallet, decimals);
+  const price = usePrice(mint.toBase58());
+
+  const contents = (
+    <>
+      <div className={styles.nameAndIcon}>
+        {image ? (
+          <img alt="" src={image} className={styles.icon} />
+        ) : (
+          <div className={styles.icon} />
+        )}
+        <div className={styles.nameAndMint}>
+          <span className={styles.name}>{name ?? mint.toBase58()}</span>
+          <CopyButton className={styles.mint ?? ""} text={mint.toBase58()}>
+            <TruncateKey keyValue={mint} />
+          </CopyButton>
+        </div>
+      </div>
+      <div className={styles.amountAndActions}>
+        <div className={styles.amountAndDetails}>
+          <span className={styles.amount}>{amountAsString}</span>
+          {price.type === PriceDataStateType.Loaded && (
+            <NotionalAmount
+              amount={amountAsString}
```

Comment:
i don't love this conversion to string and then back within NotionalAmount, but i preferred this to what was here before, which was essentially the internals of the NotionalAmount component. We could make the amount prop `string | bigint`?